src/helpers/MiscFunctions.cpp

```diff
@@ -346,14 +367,54 @@ int getWorkspaceIDFromString(const std::string& in, std::string& outName) {
             currentItem += remains;
 
             // sanitize
-            if (currentItem >= (int)validWSes.size()) {
-                currentItem = currentItem % validWSes.size();
-            } else if (currentItem < 0) {
-                currentItem = validWSes.size() + currentItem;
+            if (!includeEmpty) {
+                if (currentItem >= (int)validWSes.size()) {
+                    currentItem = currentItem % validWSes.size();
+                } else if (currentItem < 0) {
+                    currentItem = validWSes.size() + currentItem;
+                }
+            } else {
+                if (currentItem >= (int)validWSes.size()) {
```

Comment:
can't we somehow unify this, and possibly get rid of the while loop? It has caused issues in the past if someone had e.g. workspace `12378723` open

Reply:
For 'r' and 'e' we can just wrap around and call it a day. For 'r' we don't wrap around and need to consider the possibility, that someone calls e.g. r+5 at the last open workspace. So I don't see a way to unify this (Unless you mean the two direction code paths). 

Concerning the while loop. AFAIK, not an issue. For this code to be relevant, we need to be currently on e.g. ws ```123456```, we only need to find at most ```|remains|``` workspaces. As above, this will only hang if someone calls ```r+123456```

Reply:
I've thought of another way of implementing that doesn't have the mentioned issues.